### PR TITLE
[DOC] Add inline docs to stream util methods and rename chainStream to chain.

### PR DESCRIPTION
**Pull request by lukemelia**
@mixonic spin-off of your pointers about the non-stream-safe util methods from #10093

### CODE REVIEW

**Comment by mmun** [comment]
Looks good. These all need to be explicitly marked `@private` or given a global exports. I don't like the name `chainStream`. I prefer `chain` or `map`.

**Comment by lukemelia** [comment]
Thanks @mmun I'll mark them private for now and we can decide whether to expose them as public API over time. Also will rename `chainStream` to `chain` and do a little doc formatting cleanup.

**Comment by lukemelia** [comment]
OK, all set from my perspective.

File: packages/ember-metal/lib/streams/utils.js [hunk 1]
```diff
@@ -94,11 +175,42 @@ export function concat(array, key) {
...
+       return currentValue + 1;
+     });
+ ```
+
+ In the example, result is a stream if source is a stream, or a number of
+ source was numeric.
+
+ @private
+ @function chain
+ @param {Object|Stream} array An array containing zero or more stream objects
```
Comment (mmun) [inline packages/ember-metal/lib/streams/utils.js]: These aren't right.

**Comment by rwjblue** [comment]
Thanks Luke!

File: packages/ember-metal/lib/streams/utils.js (hunk 1 above)
Comment (rwjblue) [inline packages/ember-metal/lib/streams/utils.js]: D'oh!

Comment (lukemelia) [inline packages/ember-metal/lib/streams/utils.js]: Whoops, good catch. Will fix.

Comment (lukemelia) [inline packages/ember-metal/lib/streams/utils.js]: Fix in #10107

**Pull request merged**